Library/Homebrew/cmd/update.rb

```diff
@@ -181,8 +181,20 @@ def report
         next unless paths.any? { |p| File.dirname(p) == formula_directory }
 
         case status
-        when "A", "M", "D"
+        when "A", "D"
           map[status.to_sym] << repository.join(src)
+        when "M"
+          file = repository.join(src)
+          begin
+            require "formula_versions"
+            formula = Formulary.factory(file)
+            new_version = formula.pkg_version
+            old_version = FormulaVersions.new(formula).formula_at_revision(@initial_revision, &:pkg_version)
+            next if new_version == old_version
+          rescue Exception => e
```

Comment:
> What is this protecting against?

Formula loading error and possible IO error inside FormulaVersions.

> This will catch programming errors in any code invoked in the begin block, so we should avoid it if possible.

I intend to protect `brew update` command as much as possible.